README.md

```diff
@@ -0,0 +1,100 @@
+# KCP Helm Charts
+
+Repository for KCP helm charts.
+
+## Pre-requisites
+
+- Cert-manager installed and running
+- Ingress installed (e.g. nginx-ingress or OpenShift router)
+
+## Usage
+
+[Helm](https://helm.sh) must be installed to use the charts.  Please refer to
+Helm's [documentation](https://helm.sh/docs) to get started.
+
+Once Helm has been set up correctly, add the repo as follows:
+
+    ```bash
+    helm repo add kcp https://kcp-dev.github.io/helm-charts
+    ```
+
+If you had already added this repo earlier, run `helm repo update` to retrieve the latest versions of the packages.  You can then run `helm search repo kcp` to see the charts.
+
+To install the kcp chart:
+
+    ```bash
+    helm install my-kcp kcp/kcp
+    ```
+
+To uninstall the chart:
+
+    ```bash
+    helm delete my-kcp
+    ```
+
+## Development usage
+
+To install using the local chart:
+
+    ```bash
+    helm install kcp ./charts/kcp --values ./myvalues.yaml --namespace kcp --create-namespace
+    ```
+
+Changes can then be made locally and tested via upgrade:
+
+    ```bash
+    helm upgrade kcp ./charts/kcp --values ./myvalues.yaml --namespace kcp
+    ```
+
+Note `myvalues.yaml` will depend on your environment (you must specify which ingress method
+is used to expose the front-proxy endpoint), a minimal example:
+
+    ```yaml
+    externalHostname: "<external hostname as exposed by ingress method below>"
+    kcpFrontProxy:
+      ingress:
+        enabled: true
+    ```
+### Install to KIND cluster (for development)
+
+There is a helper script to install KCP to a KIND cluster. It will install cert-manager, nginx-ingress and KCP.
+Kind cluster binds to host ports 80 and 443 for ingress. Ingress is emulated using host entries in `/etc/hosts`.
+This particular configuration is useful for development and testing, but will not work with LetsEncrypt.
+
+    ```bash
+    ./hack/kind-setup.sh
+    ```
+
+Pre-requisites:
+* Kind cluster running (e.g. `kind create cluster`)
+* Cert-manager installer and running
+* Ingress installed
+* `/etc/hosts entry` for `kcp.dev.local` pointing to `127.0.0.1`
+
+Run helm install:
+
+      ```bash
+      helm upgrade -i my-kcp ./charts/kcp/ \
+      --values values.yaml \
+      --namespace kcp \
+      --create-namespace
+      ```
+
+Where `values.yaml` is:
+
+      ```yaml
+      externalHostname: "kcp.dev.local"
+      kcp:
+        volumeClassName: "standard"
+      kcpFrontProxy:
+        ingress:
+          enabled: true
+          annotations:
+            kubernetes.io/ingress.class: "nginx"
+            nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
+      ```
+
+# Known issues
+
+* https://github.com/kcp-dev/kcp/issues/2295 - Deployments fail to start.
+Workaround: Delete the empty token file in `kcp` volume and restart kcp pod.
```

Comment:
This happens almost always in my kind cluster on new deployments :/ 
@s-urbaniak did you manage to get to the bottom of this? If not we can look into my setup as it almost always replicatable

Reply:
I got to the bottom of it. You can't delete .kcp/admin.kubeconfig but preserve .kcp/.admin-token-store. We used to do that in kcp CI make targets but https://github.com/kcp-dev/kcp/pull/2707/files fixed it.

Reply:
> I got to the bottom of it. You can't delete .kcp/admin.kubeconfig but preserve .kcp/.admin-token-store. We used to do that in kcp CI make targets but https://github.com/kcp-dev/kcp/pull/2707/files fixed it.

@ncdc I'm still hitting this issue (I had to do the workaround today using the latest image from https://github.com/kcp-dev/kcp/pkgs/container/kcp) but https://github.com/kcp-dev/kcp/pull/2707 looks like it merged 2 weeks ago?

Reply:
I think it may be around [here](https://github.com/kcp-dev/kcp/blob/main/pkg/server/options/authentication.go#L99) - if the token hash file exists but the admin kubeconfig doesn't, we'll end up with an empty shardAdminToken?

Reply:
Yes, that's what I meant by you can't delete the admin.kubeconfig file by itself.